### PR TITLE
Adding coverage for make_subnet method

### LINKED ISSUE #1712: Add CLI tests for three new parameters in factory.make_subnet
new parameters
```
u'domain-ids': None,
u'gateway': None,
u'ipam': None,
```

### PR DESCRIPTION
**Pull request by oshtaier**
Cover three more parameters in order to
Close #1712 issue

### CODE REVIEW

File: tests/foreman/cli/test_subnet.py [hunk 1]
```diff
@@ -3,9 +3,15 @@
...
-from robottelo.cli.factory import make_subnet, CLIFactoryError
+from robottelo.cli.factory import make_domain, make_subnet, CLIFactoryError
 from robottelo.cli.subnet import Subnet
-from robottelo.common.decorators import bz_bug_is_open, data, run_only_on
+from robottelo.common.constants import SUBNET_IPAM_TYPES
+from robottelo.common.decorators import (
+    bz_bug_is_open,
+    data,
+    run_only_on,
+    skip_if_bug_open
```
Comment (elyezer) [inline tests/foreman/cli/test_subnet.py]: To easy maintenance please include a comma at the end.

**Comment by elyezer** [comment]
ACK, pending a nitpick (not a blocker but will be good to have it in)

File: tests/foreman/cli/test_subnet.py (hunk 1 above)
Comment (oshtaier) [inline tests/foreman/cli/test_subnet.py]: My bad. Fixed

File: robottelo/common/constants.py [hunk 2]
```diff
@@ -523,3 +523,7 @@
 
 ANY_CONTEXT = {'org': "Any Organization",
                'location': "Any Location"}
+
+SUBNET_IPAM_TYPES = {'dhcp': "DHCP",
+                     'internal': "Internal DB",
+                     'none': "None"}
```
Comment (elyezer) [inline robottelo/common/constants.py]: The above `ANY_CONTEXT` format is something old, please follow the new one:

```python
SUBNET_IPAM_TYPES = {
    'dhcp': 'DHCP',
    'internal': 'Internal DB',
    'none': 'None',
}
```

**Comment by elyezer** [comment]
Have just reviewed #2090 and came up with that would be great to have two tests for parameters that expects a list of elements. With that said I think worth creating two tests for `domain-ids`, one passing just one element and the other passing 2 or more elements (a random between 2-5 inclusive should be good).

ACK, pending suggestion and style comments.

**Comment by oshtaier** [comment]
I think mentioned suggestion is out of scope of mentioned task: 1. We have already finished around 10 tasks in similar areas (make_*) that also have lists as input parameters, but we go further for single case scenario. 2. Things like this always mentioned in the task description in advance as I see in another issues descriptions.
It was said that we need to cover parameters, but not all scenarios, because there are not only these two that you mentioned

Anyway, I am fine to proceed, but we need to change task description in that case

**Comment by elyezer** [comment]
@oshtaier we can leave this PR as is then. Can you please create more specific issues for the multi value params?

**Comment by oshtaier** [comment]
@elyezer Sure, not a problem to create new issues for multiple values, but we are closing current PR and issue, right?

**Comment by elyezer** [comment]
Yes, the multi-value is not required to close this. Let's wait by the second ACK

**Comment by Ichimonji10** [comment]
ACK

Nice job.

**Pull request merged**